tests/suite/test_custom_annotations.py

```diff
@@ -0,0 +1,88 @@
+
+    Attributes:
+        public_endpoint (PublicEndpoint):
+        namespace (str):
+        ingress_host (str):
+    """
+    def __init__(self, public_endpoint: PublicEndpoint, ingress_name, namespace, ingress_host, ic_pod_name):
+        self.public_endpoint = public_endpoint
+        self.namespace = namespace
+        self.ingress_name = ingress_name
+        self.ingress_host = ingress_host
+        self.ic_pod_name = ic_pod_name
+        self.backend_1_url = f"http://{public_endpoint.public_ip}:{public_endpoint.port}/backend1"
+        self.backend_2_url = f"http://{public_endpoint.public_ip}:{public_endpoint.port}/backend2"
+
+
+@pytest.fixture(scope="class")
+def custom_annotations_setup(request, kube_apis, ingress_controller_prerequisites,
+                             ingress_controller_endpoint, test_namespace) -> CustomAnnotationsSetup:
+    ing_type = request.param
+    print("------------------------- Deploy ConfigMap with custom template -----------------------------------")
+    replace_configmap_from_yaml(kube_apis.v1,
+                                ingress_controller_prerequisites.config_map['metadata']['name'],
+                                ingress_controller_prerequisites.namespace,
+                                f"{TEST_DATA}/custom-annotations/{ing_type}/nginx-config.yaml")
+    print("------------------------- Deploy Custom Annotations Example -----------------------------------")
+    ing_src = f"{TEST_DATA}/custom-annotations/{ing_type}/annotations-ingress.yaml"
+    create_items_from_yaml(kube_apis, ing_src, test_namespace)
+    host = get_first_ingress_host_from_yaml(ing_src)
+    ingress_name = get_names_from_yaml(ing_src)[0]
+    create_example_app(kube_apis, "simple", test_namespace)
+    wait_until_all_pods_are_ready(kube_apis.v1, test_namespace)
+
+    ic_pod_name = get_first_pod_name(kube_apis.v1, ingress_controller_prerequisites.namespace)
+
+    def fin():
+        print("Clean up Custom Annotations Example:")
+        replace_configmap_from_yaml(kube_apis.v1,
+                                    ingress_controller_prerequisites.config_map['metadata']['name'],
+                                    ingress_controller_prerequisites.namespace,
+                                    f"{DEPLOYMENTS}/common/nginx-config.yaml")
+        delete_items_from_yaml(kube_apis, ing_src, test_namespace)
+        delete_common_app(kube_apis, "simple", test_namespace)
+
+    request.addfinalizer(fin)
+
+    return CustomAnnotationsSetup(ingress_controller_endpoint, ingress_name, test_namespace, host, ic_pod_name)
+
+
+@pytest.mark.ingresses
+@pytest.mark.parametrize('custom_annotations_setup, expected_texts',
+                         [
+                             pytest.param("standard",
+                                          ["# This is TEST configuration for custom-annotations-ingress/",
+                                           "# Insert config for test-split-feature: 192.168.1.1;",
```

Comment:
here`# Insert config for test-split-feature: some-ip;` will also be generated, should we test for that?